### PR TITLE
Fixing type attribute warning for transfer_action

### PR DESCRIPTION
**Pull request by msimberg**
Retry of #3483. I didn't change anything from @sithhell's original, so I'm expecting pycicle gcc builds to fail.

### CODE REVIEW

**Comment by msimberg** [comment]
Apparently the builds on daint are ok with gcc... The errors on daint were [these](http://rostam.cct.lsu.edu/builders/hpx_gcc_7_boost_1_65_centos_x86_64_release/builds/130/steps/build_unit_tests/logs/stdio), so something is different on rostam.

**Comment by hkaiser** [comment]
> Apparently the builds on daint are ok with gcc... The errors on daint were these, so something is different on rostam.

Could it be different `--fvisibility=…` settings (or whatever the command line options are)?

**Comment by msimberg** [comment]
> > Apparently the builds on daint are ok with gcc... The errors on daint were these, so something is different on rostam.
> 
> Could it be different `--fvisibility=…` settings (or whatever the command line options are)?

I had hoped that would be the problem, but that doesn't seem to be the case. We can maybe have a look next week.

**Comment by msimberg** [comment]
I'm not sure what is going on. Manually triggered builds with the same configuration that failed before don't fail now, and nothing relevant has changed. I think I'll just have to try merging again, and retry if it fails.

File: src/components/containers/partitioned_vector/partitioned_vector_component_double.cpp [hunk 1]
```diff
@@ -21,7 +21,10 @@ HPX_REGISTER_PARTITIONED_VECTOR(double);
 #pragma warning(disable: 5037)
 #endif
 
-template class HPX_PARTITIONED_VECTOR_EXPORT
+template class
+#if defined(HPX_MSVC) || defined(HPX_MINGW)
+HPX_PARTITIONED_VECTOR_EXPORT
+#endif
```
Comment (hkaiser) [inline src/components/containers/partitioned_vector/partitioned_vector_component_double.cpp]: Could we handle that by defining a separate pp constant (similarly to `HPX_PARTITIONED_VECTOR_SPECIALIZATION_EXPORT`, see here: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/components/containers/partitioned_vector/export_definitions.hpp#L17-L21)?

Comment (msimberg) [reply src/components/containers/partitioned_vector/partitioned_vector_component_double.cpp]: Ooh, yes, absolutely. Thanks for pointing that out.

**Comment by hkaiser** [comment]
@sithhell, @msimberg what's the status here. Can we go ahead and merge this now?

**Pull request merged**